curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641da3c6b6fbd742bff6ee40.md

```diff
@@ -0,0 +1,138 @@
+---
+id: 641da3c6b6fbd742bff6ee40
+title: Step 2
+challengeType: 0
+dashedName: step-2
+---
+
+# --description--
+
+You will now get into using the `Fetch API`. It is a built-in JavaScript interface for getting a resource from a server. It has a `fetch()` method with which you can make a `GET`, `POST`, `PUT`, or `PATCH` request.
+
+The request you are going to make in this project is a `GET` request to this URL `https://cdn.freecodecamp.org/curriculum/news-author-page/authors.json`. The URL has the information of freeCodeCamp News Site authors – making it ideal for what you want to build.
+
+To make a `GET` request with the `fetch()` method, you do it like this:
+
+```js
+fetch("sample-url-goes-here")
+```
+
+Make a `GET` request to the freeCodeCamp authors URL. Don't terminate the code with a semi-colon yet.
+
+# --hints--
+
+You should use the `fetch()` method to make a `GET` request to `https://cdn.freecodecamp.org/curriculum/news-author-page/authors.json`.
+
+```js
+assert.match(code, /fetch\(('|"|`)https:\/\/cdn\.freecodecamp\.org\/curriculum\/news\-author\-page\/authors\.json\1\)/)
+```
```

Comment:
````suggestion
You should use the `fetch()` method to make a `GET` request to `https://cdn.freecodecamp.org/curriculum/news-author-page/authors.json`.

```js
assert.match(code, /fetch\(('|"|`)https:\/\/cdn\.freecodecamp\.org\/curriculum\/news\-author\-page\/authors\.json\1\)/)
```

You should not have a semi-colon after your `fetch` call.

```js
assert.notMatch(code, /fetch\(('|"|`)https:\/\/cdn\.freecodecamp\.org\/curriculum\/news\-author\-page\/authors\.json\1\);/)
```
````

If we are instructing campers not to use a semi-colon, we should test for that.